polyfill/test/Calendar/prototype/fields/argument-throws-invalid-keys.js

```diff
@@ -0,0 +1,30 @@
+// Copyright (C) 2021 Igalia, S.L. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-temporal.calendar.prototype.fields
+info: |
+    sec-temporal.calendar.prototype.fields step 7.b.ii:
+      7. Repeat, while next is not false,
+        a. Set next to ? IteratorStep(iteratorRecord).
+        b. If next is not false, then
+          i. Let nextValue be ? IteratorValue(next).
+          ii. If Type(nextValue) is not String, then
+            1. Let completion be ThrowCompletion(a newly created TypeError object).
+            2. Return ? IteratorClose(iteratorRecord, completion).
+    sec-temporal.calendar.prototype.fields step 7.b.iv:
+          iv. If _nextValue_ is not one of *"year"*, *"month"*, *"monthCode"*, *"day"*, *"hour"*, *"minute"*, *"second"*, *"millisecond"*, *"microsecond"*, *"nanosecond"*, then
+            1. Let _completion_ be ThrowCompletion(a newly created *RangeError* object).
+            2. Return ? IteratorClose(_iteratorRecord_, _completion_).
+
+features: [Temporal]
+---*/
+
+const calendar = new Temporal.Calendar("iso8601");
+assert.throws(TypeError, () => calendar.fields([1]));
```

Comment:
Consider being exhaustive with all of the non-string types here: `1n`, `Symbol('foo')`, etc.